app/models/heart_rate_data.rb

```diff
@@ -0,0 +1,30 @@
+class HeartRateData < ApplicationRecord
+
+  belongs_to :user, dependent: :destroy
+
+  def update_heart_rate_data!
+    update!(data: heart_rate_data_for_the_day)
+  end
+
+  private
+
+  def heart_rate_data_for_the_day
+    client.heart_rate_intraday_time_series(
+      user_id: '-',
+      date: 'today',
+      start_time: '00:00',
+      end_date: '24:00',
+      detail_level: '1min'
+    )
+  end
+
+  def client
+    client = Fitbit::Client.new(
```

Comment:
unused local variable `client` . This also may be able to be memoized if token refresh happens automagically